packages/astro/scripts/shiki-gen-themes.mjs

```diff
@@ -2,18 +2,36 @@ import fs from 'fs';
 
 const dir = await fs.promises.readdir('packages/astro/node_modules/shiki/themes/');
 
+const toThemeImport = (theme) => `import('shiki/themes/${theme}').then(mod => mod.default)`;
+
 const themeImports = dir.map((f) => {
-	return [f.slice(0, f.indexOf('.json')), `import('shiki/themes/${f}').then(mod => mod.default)`];
+	return [f.slice(0, f.indexOf('.json')), toThemeImport(f)];
 });
 
-let code = `export const themes = {`;
-let i = 0;
+// Map of old theme names to new names to preserve compatibility when we upgrade shiki
+const compatThemes = {
+	'material-darker': 'material-theme-darker',
+	'material-default': 'material-theme',
+	'material-lighter': 'material-theme-lighter',
+	'material-ocean': 'material-theme-ocean',
+	'material-palenight': 'material-theme-palenight',
+};
+
+let code = `\
+/**
+ * This file is prebuilt from packages/astro/scripts/shiki-gen-themes.mjs
+ * Do not edit this directly, but instead edit that file and rerun it to generate this file.
+ */
+
+// prettier-ignore
+export const themes = {`;
```

Comment:
Also used `// prettier-ignore` here as the codegen code is more readable without prettier's line breaks.